x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -51,7 +50,12 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
       overwrite: true,
     });
 
-    return toSLO(savedSLO.attributes);
+    const result = this.toSLO(savedSLO.attributes);
+    if (result === undefined) {
+      throw new Error('Invalid stored SLO');
+    }
```

Comment:
i find it weird that first we try to save the SLO and then we throw an error based on what's being returned from the result. 
shouldn't we throw an error before trying to save an invalid SLO

Reply:
As discussed offline, this case is only to please TS but technically would never happens since the encoding part when storing the definition would fail. Let me check if we can just return the SLO provided to the function instead of decoding the result from the so.create() function: technically they are the same

Reply:
I've removed the need to decode the response from the create call. Easier like this, and less confusing 👍🏻